packages/cpp/src/browser/cpp-client-contribution.ts

```diff
@@ -15,6 +18,9 @@ export class CppClientContribution extends BaseLanguageClientContribution {
     readonly id = CPP_LANGUAGE_ID;
     readonly name = CPP_LANGUAGE_NAME;
 
+    @inject(MessageService) private readonly messageService: MessageService;
+    @inject(ILogger) private readonly logger: ILogger;
```

Comment:
This should be using constructor injection, could you add a constructor and inject it here ?

Reply:
Also while I understand our discussion yesterday about private /protected but in this case I don't see why we could keep the logger or messageservice unaccessible by possible dependents ?

Reply:
Extenders can make their own so there's no value in exposing it. If this class doesn't need it anymore, it won't be able to remove it without breaking the API. The fact that this class uses a logger and message service is an implementation detail. Perhaps it would be useful if this class created its own value-added logger or message service and would like to expose this to its clients but that's not the case right now.

Reply:
>Extenders can make their own so there's no value in exposing it

An extender will need to inject it in it's extended class to pass it to the parent anyway. Removing it will also break the extender anyway since the constructor will be modified.  So there's no point in having this private.

We did have a discussion about that with the inversify project (that the extended class should not have to know about the implementation details of it's parent).

I remeber we decided to stick to constructor injection then but I don't remember the details maybe @akosyakov has an idea?

In the meantime I suggest we follow how theia's code is structured ATM and fix this issue separately.

Reply:
OK, sure!